broken-app/app.js

```
@@ -1,18 +1,48 @@
 const express = require('express');
-let axios = require('axios');
-var app = express();
+const app = express();
+const axios = require('axios');
+const ExpressError = require('./expressError');
 
-app.post('/', function(req, res, next) {
-  try {
-    let results = req.body.developers.map(async d => {
-      return await axios.get(`https://api.github.com/users/${d}`);
-    });
-    let out = results.map(r => ({ name: r.data.name, bio: r.data.bio }));
+app.use(express.json());
 
-    return res.send(JSON.stringify(out));
-  } catch {
-    next(err);
-  }
+app.post('/', (req, res) => {
+  if (!req.body.developers) throw new ExpressError('Developer username(s) required', 400);
+  const devInfo = req.body.developers.map(async (dev) => {
+    try {
+      const resp = await axios.get(`https://api.github.com/users/${dev}`);
+      if (resp.status !== 404) {
+        return {
+          bio: resp.data.bio,
+          name: resp.data.name,
+        };
+      }
+      
+    } catch {
+      return {message: "Username not found"};
+    }
+  });
+  Promise.all(devInfo).then((data) => {
+    return res.status(200).json(data);
+  })
+});
+
+app.use(function (req, res, next) {
+  return new ExpressError('Not Found', 404);
+});
+
+app.use((err, req, res, next) => {
+  let status = err.status || 500;
+  let message = err.message;
+
+  return res.status(status).json({
+    error: message,
+  });
 });
 
-app.listen(3000);
+app.listen(3000, function (err) {
+  if (err) {
+    console.log(`Error starting server: ${err}`);
```

Should probably raise an exception here too.